lib/tasks/alerts/create_alert_subscription_events.rake

```diff
@@ -1,11 +1,31 @@
 namespace :alerts do
   desc 'Run alert subscription events job'
-  task create_alert_subscription_events: [:environment] do
+  task create_alert_subscription_events_weekly: [:environment] do
     puts Time.zone.now
     schools = School.active
     schools.each do |school|
-      puts "Running alert subscription events for #{school.name}"
-      Alerts::GenerateSubscriptionEvents.new(school).perform
+      puts "Running WEEKLY alert subscription events for #{school.name}"
+      school.alerts.weekly.latest.each do |alert|
+        Alerts::GenerateSubscriptionEvents.new(school, alert).perform
+      end
+    end
+    puts Time.zone.now
+  end
+
+  task create_alert_subscription_events_termly: [:environment] do
+    puts Time.zone.now
+    schools = School.active
+    schools.each do |school|
+      if school.holiday_approaching?
```

Comment:
To confirm, we're relying on this doing a 7 day check for an approaching holiday and we're running this cron task every 7 days so it should only get run once per "holiday approaching" and therefore we get termly emails for whatever the "latest" alert was (and those alerts may be ones we previously thought as "termly" but get run every day now)